src/Middleware/AuthenticationMiddleware.php

```diff
@@ -189,6 +152,24 @@ protected function getAuthenticationService($request, $response)
 
             throw new RuntimeException($message);
         }
+        $forwardKeys = ['identityAttribute', 'unauthenticatedRedirect', 'queryParam'];
+        foreach ($forwardKeys as $key) {
+            $value = $this->getConfig($key);
+            if ($value) {
+                deprecationWarning(
+                    "The `{$key}` configuration key on AuthenticationMiddleware is deprecated. " .
+                    "Instead set the `{$key}` on your AuthenticationService instance."
+                );
+                if (method_exists($subject, 'setConfig')) {
```

Comment:
Maybe this should check for a concrete `AuthenticationService` implementation (`$subject instanceof AuthenticationService`) instead?